### ISSUE TITLE
Can't build with nixpkgs master?..

### ISSUE THREAD

**Issue by pshirshov** [thiagokokada/nix-alien#34]
It seems like `nix-alien` fails to build with `nixpkgs/master`:

```
error: builder for '/nix/store/afmw0vxjpzc6h66dbv35pfx6d3mkwd6c-python3.9-pyparsing-2.4.7.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       >
       For full logs, run 'nix log /nix/store/afmw0vxjpzc6h66dbv35pfx6d3mkwd6c-python3.9-pyparsing-2.4.7.drv'.
error: builder for '/nix/store/cz1rxqq0db8a4z8svmxf0dfyfdsnzxvq-python3.9-typing-extensions-3.10.0.2.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       >
       For full logs, run 'nix log /nix/store/cz1rxqq0db8a4z8svmxf0dfyfdsnzxvq-python3.9-typing-extensions-3.10.0.2.drv'.
error: 1 dependencies of derivation '/nix/store/l7ary7ycww3jhx07gn2s4l726k9zyjhi-python3.9-setuptools-scm-7.0.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q03abknx97z2cllzyd3fihk7lmw5dqr1-python3.9-poetry-core-1.0.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dbxkwn151mwgjg1anpi6bf5bpqfy6fgn-python3.9-nix-alien-0.1.0.drv' failed to build
```

Related to [#193708](https://github.com/NixOS/nixpkgs/issues/193708)

**Comment by thiagokokada** [comment]
PRs are welcome. There is one reason I pin the nixpkgs version in this repo.

**Comment by pshirshov** [comment]
Oh, I don't have enough expertise in Nix for a P/R, I'm just a user.

**Comment by hissssst** [comment]
I am experiencing exactly the same issue

```
building the system configuration...
error: builder for '/nix/store/afmw0vxjpzc6h66dbv35pfx6d3mkwd6c-python3.9-pyparsing-2.4.7.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       >
       For full logs, run 'nix log /nix/store/afmw0vxjpzc6h66dbv35pfx6d3mkwd6c-python3.9-pyparsing-2.4.7.drv'.
error: builder for '/nix/store/cz1rxqq0db8a4z8svmxf0dfyfdsnzxvq-python3.9-typing-extensions-3.10.0.2.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       >
       For full logs, run 'nix log /nix/store/cz1rxqq0db8a4z8svmxf0dfyfdsnzxvq-python3.9-typing-extensions-3.10.0.2.drv'.
error: 1 dependencies of derivation '/nix/store/l7ary7ycww3jhx07gn2s4l726k9zyjhi-python3.9-setuptools-scm-7.0.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q03abknx97z2cllzyd3fihk7lmw5dqr1-python3.9-poetry-core-1.0.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dbxkwn151mwgjg1anpi6bf5bpqfy6fgn-python3.9-nix-alien-0.1.0.drv' failed to build
```

**Comment by hissssst** [comment]
I think it is related to poetry somehow. @thiagokokada what do you think?

**Comment by hissssst** [comment]
By the way, I am not building on master, but on `unstable`, as the nix-alien's `flake.nix` defines

**Comment by thiagokokada** [comment]
> By the way, I am not building on master, but on `unstable`, as the nix-alien's `flake.nix` defines

But you're still overriding it right? Like I said, there is a reason the flake has a lockfile.

**Comment by thiagokokada** [comment]
The issue does seem related to `poetry2nix`: https://github.com/nix-community/poetry2nix/pull/736

Will need for them to fix it and them I can try to bump and see if it works.

**Comment by hissssst** [comment]
The issue has been fixed. Bump?

**Comment by thiagokokada** [comment]
> The issue has been fixed. Bump?

Doesn't seem so:

```
nix build
error: builder for '/nix/store/hmy2x2jsr8i1cn0xbh0nng20jailb5sp-python3.9-pylddwrap-1.2.1.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       >
       For full logs, run 'nix log /nix/store/hmy2x2jsr8i1cn0xbh0nng20jailb5sp-python3.9-pylddwrap-1.2.1.drv'.
error: 1 dependencies of derivation '/nix/store/qvlwbry7bzv758flqc0xsvhddb61g0hb-python3.9-nix-alien-0.1.0.drv' failed to build
```

**Comment by thiagokokada** [comment]
Huh... I see. Now I need to explicitly declare `buildInputs = [ final.setuptools ]` to `pylddwrap`.

**Comment by thiagokokada** [comment]
Merged #36. Issue should have been fixed.

If not, please say so and I will reopen the issue.
